### DISCUSSION TITLE
Full restore from SQL dump

### DISCUSSION THREAD

**Discussion by sebket** [magicbug/Cloudlog#1794] [Q&A]
Hi everyone,
I have lost my complete Cloudlog installation. All I do have is a recent backup in the form of an SQL dump file.
I tried to setup a fresh install of Cloudlog and then fill the created database with the contents of the backup via the restore function in phpmyadmin. But it doesn't seem to work, Cloudlog is not working anymore after.
Restoring the database before running the Cloudlog setup routine won't work either, setup will complain that the tables are already existing and aborts.

What is the correct way to restore Cloudlog from an SQL dump file?

I do have a backup as an adif file as well, but it doesn't contain all the info about my various station locations, so that won't work either.

If anyone could tell me what to do I'd be really grateful.

Sebastian

**Comment by phl0** [comment]
I would install Cloudlog. Then drop all the tables and import the SQL dump.
Should be important to mention that this will probably only work with exactly the same version that the dump was pulled from.

**Comment by sebket** [comment]
Thanks for your reply.
That might be difficult, as I do not know when I last updated my installation.
Perhaps in the end it's easier to split the adif file according to "my grid square", then setup a virgin installation of Cloudlog, create all the station locations again manually and then import the adif files one after another into the corresponding station location.
73 Sebastian DL2KSE

**Reply by phl0** [reply]
Doesn't have to be the exact version. But I have an idea. You need to have a look at the version number in migration table to determine which git version you should use. Then bring your git checkout to that commit, import the SQL and git pull to the latest version afterwards. I think this is very feasible. If you have a remote installation and would need support drop me an email -> DF2ET.

**Comment by magicbug** [comment]
Should be safe enough to download the files, setup application/config/config.php and database.php and point it at the database the migration system should bring it up to date.

**Reply by sebket** [reply]
I just found out that the SQL dump didn't dump the contents of some tables, including the contacts table, and that the adif file contains only the qso data from my main station location. So my log is lost.
Thank you anyway for your help!

**Reply by phl0** [reply]
Hmm then it is not recoverable I think. Bad luck :-(

**Reply by AndreasK79** [reply]
@sebket Sorry to hear. That's always a bad thing. Did you use LoTW? In that case, the logs should be recoverable from there.

**Reply by sebket** [reply]
Unfortunately no LoTW was used.
I tried a software called Recuva to try to recover the SQL data file on the drive, but didn't succeed. As you say, bad luck - the log is gone.

**Reply by phl0** [reply]
So a kind tip (which does not help here any more): Establish a regular backup. In my case I have a cron job running daily just dumping the whole SQL database.

**Reply by sebket** [reply]
I did do that, but the dump file contains only the structure, but no data...
I know, always verify your backups. That's the part I've omitted, unfortunately...
Again, thanks for your help!

**Reply by magicbug** [reply]
Doh! we learn from these things, hopefully you can rebuild from ADIFs!

**Reply by sebket** [reply]
Unfortunately not, as the adif I exported only contains the QSOs from my main station location, but not those from my 2nd QTH or my portable operations.
Yes, mistakes were made...

**Reply by sebket** [reply]
Just wanted to update you and let you know that with a combination of an older SQL dump I've still had on one of my backup drives and my activity on eQSL I was able to fully reconstruct my log.
Thanks again for your help!

**Reply by phl0** [reply]
congrats :)

**Reply by AndreasK79** [reply]
Good to hear!
